### PR TITLE
Update team lists and separate structure from governance

### PR DESCRIPTION
**Pull request by choldgraf**
This restructures our team and governance page a little bit to make the information more cleanly-separated. Here are the major changes.

- We now define governance in a separate location from "teams and their responsibilities"
- Updated team privileges to reflect our current GitHub org permissions
- Now auto-populating a list of team members based on our organization's github teams and documents this practice.
  - This must be manually run for now, but in the future we should automate it. Didn't want to make perfect the enemy of good-enough for now though.

### CODE REVIEW

**Review by gregcaporaso — APPROVED** [review]
@choldgraf, I reviewed and this all looks good to me. I approve merging.

File: docs/conf.py [hunk 1]
```diff
@@ -53,7 +53,7 @@
 # Functions that this calls are below
 def setup(app: Sphinx):
     app.connect("builder-inited", update_contributing)
-
+    app.add_crossref_type("team", "team")
```
Comment (rowanc1) [inline docs/conf.py:56]: So cool. :)

Comment (choldgraf) [reply docs/conf.py:56]: I just discovered this yesterday and am going to over use the shit out of it for a while I suspect haha

**Comment by choldgraf** [comment]
Thanks @gregcaporaso - this doesn't change policy at all, just updates it to be in line with current practices, so I'll merge this now that we have an approval!

**Pull request merged**